src/Server/WebSocketServer.php

```diff
@@ -162,10 +164,51 @@ private function onNewConnection(ConnectionInterface $socketStream)
             return $this->getMessageHandler($uri, $connection);
         }, $this->loop, $this->messageProcessor);
 
+        $socketStream->on('end', function () use($connection) {
```

Comment:
This should probably be in `Connection:: initListeners`

Reply:
You can give a callback as parameter to the connection that will occur on disconnection if you want.